File: 74-SearchA2dMatrix/74-SearchA2dMatrix.py
# Last updated: 2/3/2026, 9:42:58 PM
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n=len(matrix)
        m=len(matrix[0])
        top,bot=0,n*m-1
        while top<=bot:
            mid=(top+bot)//2
            row=mid//m
            col=mid%m
            if matrix[row][col]==target:
                return True
            elif matrix[row][col]<target:
                top=mid+1
            else:
                bot=mid-1
        return False

File: 74-SearchA2dMatrix/74-SearchA2dMatrix.java
// Last updated: 2/3/2026, 9:43:05 PM
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
            int top=0;
            int bottom =matrix.length-1;

            while(top<=bottom){
            int mid=(top+bottom)/2;
            if (matrix[mid][0] < target && matrix[mid][matrix[mid].length - 1] > target) break;
            else if(target<matrix[mid][0]) bottom=mid-1;
            else top=mid+1;
            }
            int row=(top+bottom)/2;
            int left=0;
            int right=matrix[row].length-1;
            while(left<=right){
                int mid=(left+right)/2;
                if(matrix[row][mid]==target) return true;
                else if(target<matrix[row][mid]) right=mid-1;
                else left=mid+1;
            }
        return false;

    }
}